pennylane/ops/functions/dot.py

```diff
@@ -42,13 +44,25 @@ def dot(
             operator is used to represent the linear combination. If False, a :class:`Sum` operator
             is returned. Defaults to ``False``. Note that when ``ops`` consists solely of ``PauliWord``
             and ``PauliSentence`` instances, the function still returns a PennyLane operator when ``pauli=False``.
+        grouping_type (str): The type of binary relation between Pauli words used to compute
+            the grouping. Can be ``'qwc'``, ``'commuting'``, or ``'anticommuting'``. Note that if
+            ``pauli=True``, the grouping will be ignored.
+        method (str): The graph coloring heuristic to use in solving minimum clique cover for
+            grouping, which can be ``'lf'`` (Largest First) or ``'rlf'`` (Recursive Largest
+            First). This keyword argument is ignored if ``grouping_type`` is ``None``.
```

Comment:
Unless I've missed it, there's nowhere in the docs where we clearly describe (or give an external reference to) some of the terminology here. Any suggestion on how to tackle this (not necessarily in this PR)? Perhaps we could link to [here](https://docs.pennylane.ai/en/stable/code/qml_pauli.html#module-pennylane.pauli.grouping.graph_colouring) and give a few more details if needed.

Reply:
Yes good point, I think your link to the module description as well as the `group_observables` docs are good pointers. Though now that I see it again I think we should at least write out somewhere what "qwc" stands for

Reply:
There is also this demo by Josh https://pennylane.ai/qml/demos/tutorial_measurement_optimize/

Reply:
Great, how about:
- In this docstring we link to the further details in the `pauli` module docstring
- The `pauli` module docstring can somewhere point to the measurement optimization demo for more details.

Reply:
sounds good to me 👍